lib/OpenQA/WebAPI/Plugin/ObsRsync/Task.pm

```diff
@@ -52,14 +49,14 @@ sub run {
     return $job->retry({delay => $retry_interval}) if $dirty;
 
     return $job->retry({delay => $retry_interval})
-      unless my $concurrency_guard = $minion->guard('obs_rsync_run_guard', $lock_timeout, {limit => $concurrency});
+      unless my $concurrency_guard = $helper->concurrency_guard();
 
     $helper->log_job_id($project, $job->id);
     my @cmd = (Mojo::File->new($home, 'script', 'rsync.sh')->to_string, $project);
     my ($stdin, $stdout, $error);
-    IPC::Run::run(\@cmd, \$stdin, \$stdout, \$error);
-    my $exit_code = $?;
-    $minion->unlock('obs_rsync_project_' . $project . '_lock');
+    my $exit_code = -1;
+    eval { IPC::Run::run(\@cmd, \$stdin, \$stdout, \$error); $exit_code = $?; };
```

Comment:
there seems to be no actual handling of exceptions

Reply:
Exist code and error are logged below, so I suppose it is ok.